src/__tests__/globalStateE4.test.ts

```diff
@@ -0,0 +1,44 @@
+import { expect, test, describe } from "bun:test";
+import { generatePatch } from "../patches/globalStateE4";
+import { Patch, type Circuit } from "../patch";
+
+describe("GlobalStateE4", () => {
+  test("generates patch with valid circuit keys", () => {
+    const patch = new Patch();
```

Comment:
no, we want to check `const patch = generatePatch()` and inside patch has no not allowed keys